rossum_api/clients/internal_sync_client.py

```diff
@@ -0,0 +1,294 @@
+    def __init__(
+        self,
+        base_url: str,
+        credentials: UserCredentials | Token,
+        *,
+        timeout: Optional[float] = None,
+        n_retries: int = 3,
+        response_post_processor: Optional[ResponsePostProcessor] = None,
+    ):
+        self.base_url = base_url
+        self.client = httpx.Client(timeout=timeout)
+        self.n_retries = n_retries
+        self.response_post_processor = response_post_processor
+
+        self.token = None
+        self.username = None
+        self.password = None
+        if isinstance(credentials, UserCredentials):
+            self.username = credentials.username
+            self.password = credentials.password
+        else:
+            self.token = credentials.token
+
+    def _authenticate(self) -> None:
+        response = self.client.post(
+            f"{self.base_url}/auth/login",
+            data={"username": self.username, "password": self.password},
+        )
+        if self.response_post_processor is not None:
+            self.response_post_processor(response)
+        self._raise_for_status(response)
+        self.token = response.json()["key"]
+
+    @property
+    def _headers(self):
+        return {"Authorization": f"token {self.token}"}
+
+    def create(self, resource: Resource, data: dict[str, Any]) -> dict[str, Any]:
+        """Create a new object."""
+        return self.request_json("POST", resource.value, json=data)
+
+    def replace(self, resource: Resource, id_: int, data: dict[str, Any]) -> dict[str, Any]:
+        """Modify an entire existing object."""
+        return self.request_json("PUT", f"{resource.value}/{id_}", json=data)
+
+    def update(self, resource: Resource, id_: int, data: dict[str, Any]) -> dict[str, Any]:
+        """Modify particular fields of an existing object."""
+        return self.request_json("PATCH", f"{resource.value}/{id_}", json=data)
+
+    def delete(self, resource: Resource, id_: int) -> None:
+        """Delete a particular object.
+
+        Use with caution: For some objects, it triggers a cascade delete of related objects.
+        """
+        self._request("DELETE", f"{resource.value}/{id_}")
+
+    def upload(
+        self,
+        url: str,
+        files: dict[str, Any],
+    ) -> dict[str, Any]:
+        """Upload a file to a resource that supports this."""
+        return self.request_json("POST", url, files=files)
+
+    @staticmethod
+    def _build_export_query_params(
+        export_format: str,
+        columns: Sequence[str] = (),
+        **filters: Any,
+    ):
+        query_params = {"format": export_format}
+        filters = filters or {}
+        if filters:
+            query_params = {**query_params, **filters}
+        if columns:
+            query_params["columns"] = ",".join(columns)
+        return query_params
+
+    def export(
+        self,
+        resource: Resource,
+        id_: int,
+        export_format: str,
+        http_method: Any,
```

Comment:
I see a small discrepancy in the sync/async API. Async version uses `get_http_method_for_annotation_export` to infer the method and here we force user to pass it.